ament_cmake_vendor_package/cmake/ament_vendor.cmake

```diff
@@ -0,0 +1,265 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+macro(ament_vendor TARGET_NAME)
+  if(NOT PROJECT_NAME)
+    message(FATAL_ERROR "ament_vendor() must be called after project()")
+  endif()
+
+  if(_${PROJECT_NAME}_AMENT_PACKAGE)
+    message(FATAL_ERROR "ament_vendor() must be called before ament_package()")
+  endif()
+
+  cmake_parse_arguments(_ARG "GLOBAL_HOOK;SKIP_INSTALL" "SOURCE_SUBDIR;VCS_TYPE;VCS_URL;VCS_VERSION;SATISFIED" "CMAKE_ARGS;PATCHES" ${ARGN})
```

Comment:
Mind adding a comment [like this one](https://github.com/ament/ament_cmake/blob/daf0ddf9424020a3bd90243b468a0fdc1cfce4bb/ament_cmake_core/cmake/index/ament_index_get_resource.cmake#L15-L30)? I looked through all the example implementations and had a couple questions about the arguments

* What does `GLOBAL_HOOK` mean? It seems to create environment hooks - is it creating environment hooks to the installed vendor package such that the vendored packages could be used without `find_package()`ing the `*_vendor` package? (assuming there's a `<depend>*_vendor</...` so the package is accessible in an isolated workspace.
* Why do `CMAKE_ARGS` get passed what [looks like cxx flags](https://github.com/ros2/console_bridge_vendor/compare/rolling...cottsay/ament_vendor#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR23-R26)? Maybe these are CMake flags and I didn't know it, or maybe there's a bug here where `-DCMAKE_CXX_FLAGS=` is missing.

Reply:
> What does GLOBAL_HOOK mean?

Precisely. There are several instances where we're implicitly requiring this today. We should eventually deprecate this option or make it the default - a discussion for a later date. For now, it's needed in some places but not all.

> Why do CMAKE_ARGS get passed what [looks like cxx flags](https://github.com/ros2/console_bridge_vendor/compare/rolling...cottsay/ament_vendor#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR23-R26)?

It's a bug - I'll fix it. Nice catch.

Reply:
> Mind adding a comment [like this one](https://github.com/ament/ament_cmake/blob/daf0ddf9424020a3bd90243b468a0fdc1cfce4bb/ament_cmake_core/cmake/index/ament_index_get_resource.cmake#L15-L30)?

Done, see 506dd7c